lib/subtitle.rb

```diff
@@ -68,6 +73,66 @@ def transform(types, options = nil, target_lang = nil, src_lang = nil)
     @handler.transform_to(vals, src_lang, target_lang, output_dir)
   end
 
+  def generate_srt(outfile)
+    srt_helper = SrtHelper.new
+    srt_helper.parse_file(@cc_file, outfile)
+  end
+
+  def transcribe(options)
+    # From here call the "aws.rb"
+    # Get the transcribe output json file 
+    # Pass it to the generate_srt
+
+    aws = AwsEngine.new(options)
+    outfile = options[:outfile]
+    bucket = options[:bucket]
+    audio_lang = options[:audio_lang]
+    video_file = options[:video_file]
+    transcribe_json_file = nil
+    isurl = isURL?(video_file)
+    if(isurl == true)
+      output = aws.transcribe_uri(video_file, audio_lang, bucket)
+    else
+      output = aws.transcribe_file(video_file, audio_lang, bucket)
+    end
+    if output["status"].eql?("FAILED")
+      failure_reason = output["failure_reason"]
+      #return "AWS Transcribe failed #{failure_reason}"
+    end
+    transcribe_json_file = output["temp_json_output"]
+    srt_helper = SrtHelper.new
+    srt_helper.parse_file(transcribe_json_file, outfile)
+    File.delete(transcribe_json_file) if File.exist?(transcribe_json_file)
+    aws.delete_temp_transcribe_file(bucket, output)
+  end
+
+  def isURL?(url_string)
```

Comment:
Do we need all these URL logics considering transcribe only accepts s3 urls at the moment ?